test/initial_tfm.jl

```diff
@@ -0,0 +1,211 @@
+    @test !isrotation(SD*tformtest6.linear*inv(SD))
+
+    tformtest66, mm66 = RegisterQD.qd_rigid_coarse(testimage6, testimage5, mxshift, mxrot2, minwidth_rot2; SD=SD, print_interval=typemax(Int), initial_tfm = tformtest6)
+    @test mm66 <1e-4
+    @test isrotation(tformtest66.linear)
+    @test !isrotation(SD*tformtest66.linear*inv(SD))
+
+    tformtest7, mm7 = RegisterQD.qd_rigid_fine(testimage6, testimage5, [mxrot2...]./2, minwidth_rot2; SD=SD, print_interval=typemax(Int))
+    @test mm7 <1e-4
+    @test isrotation(tformtest7.linear)
+    @test !isrotation(SD*tformtest7.linear*inv(SD))
+
+    tformtest77, mm77 = RegisterQD.qd_rigid_fine(testimage6, testimage5, [mxrot2...]./2, minwidth_rot2; SD=SD, print_interval=typemax(Int), initial_tfm = tformtest7)
+    @test mm77 <1e-4
+    @test isrotation(tformtest77.linear)
+    @test !isrotation(SD*tformtest77.linear*inv(SD))
+    #fails due to specific error in qd_rigid_fine
+
+    # a non rigid initial_tfm does not return a rigid transformation and prints an error message (?)
+    tformtest8, mm8 = qd_rigid(testimage6, testimage5, mxshift, mxrot2; SD=SD, print_interval=typemax(Int), initial_tfm = AffineMap(SD\tformtest5.linear*SD, SD\tformtest5.translation))
+    @test mm8 <1e-4
+    @test isapprox(tformtest8, mytform, atol = 1)
+    @test !isrotation(tformtest8.linear)
+
+    mktemp() do path, io
+        redirect_stdout(io) do
+            tformtest8, mm8 = qd_rigid(testimage6, testimage5, mxshift, mxrot2; SD=SD, print_interval=typemax(Int), initial_tfm = AffineMap(SD\tformtest5.linear*SD, SD\tformtest5.translation))
+        end
+        flush(io)
+        str = read(path, String)
+        @test !isempty(str)
+    end
+
+end #TODO why do these pass when they're missing variables?
+
+@testset "Test initial_tfm improves translational alignment for affine" begin
+    testimage3 = zeros(10,20,10)
+    testimage3 .= NaN
+    testimage3[3:8,3:8,3:8] = gradcube
+    testimage3 = centered(testimage3)
+
+    testimage4 = zeros(10,20,10)
+    testimage4 .= NaN
+    testimage4[3:8,13:18,3:8] = gradcube
+    testimage4 = centered(testimage4)
+
+    init_tfm = AffineMap(diagm(0 => fill(1.0,3)), [0.0,10.0,0.0])
+
+    tformtest3 = nothing
+    mm3 = nothing
+
+    @test try
+        tformtest3, mm3 = qd_affine(testimage3, testimage4, mxshift; print_interval=typemax(Int)) #
+        false
+    catch err
+        true  # this should break
+    end
+
+    tformtest4, mm4 = qd_affine(testimage3, testimage4, mxshift;  print_interval = typemax(Int), initial_tfm = init_tfm, fvalue = 1e-5) #
+    @test isapprox(mm4, 0, atol = 0.0001) #this mismatch should be lower!
+    @test isapprox(tformtest4.translation, [0.0, 10.0, 0.0])
+
+end
+
+@testset "Test initial_tfm improves general alignment for affine" begin
+    testimage1 = zeros(Float64,10,10,10)
+    testimage1 .= NaN
+    testimage1[3:8,3:8,3:8] = gradcube
+    testimage1 = centered(testimage1)
+
+    tform = RotXYZ(0.1,0.1,0.1)
```

Comment:
You could even add some skew here perhaps. Not essential though.

Reply:
As other tests incorporate skew, I would prefer to just test the effects of initial tfm without worrying about other ways I could exceed the default parameters in this testset.